apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -679,45 +716,96 @@ contract DisputableVoting is IForwarderWithContext, DisputableAragonApp {
+    *      Assumes all eligibility checks have passed for the given vote and voter
+    * @param _vote Vote instance
+    * @param _voteId Identification number of vote
+    * @param _supports Whether principal voter supports the vote
+    * @param _voter Address of principal voter
+    * @param _caster Address of vote caster, if voting via representative
     */
-    function _voteOnBehalfOf(uint256 _voteId, bool _supports, address[] _voters) internal {
-        Vote storage vote_ = _getVote(_voteId);
-        // Note that the period for representatives to vote can never go into a quiet ending
-        // extension, and so we don't need to check other timing-based pre-conditions
-        require(_canRepresentativesVote(vote_), ERROR_PAST_REPRESENTATIVE_VOTING_WINDOW);
+    function _castVote(Vote storage _vote, uint256 _voteId, bool _supports, address _voter, address _caster) internal {
+        Setting storage setting = settings[_vote.settingId];
+        // Note: if we move the quiet ending check at the end here, we could optimize this so it doesn't get calculated outside of the quiet ending period
+        bool wasAccepted = _isAccepted(_vote, setting);
 
-        for (uint256 i = 0; i < _voters.length; i++) {
-            address voter = _voters[i];
-            require(_hasVotingPower(vote_, voter), ERROR_CANNOT_VOTE);
-            require(_isRepresentativeOf(voter, msg.sender), ERROR_NOT_REPRESENTATIVE);
+        uint256 yeas = _vote.yea;
+        uint256 nays = _vote.nay;
+        uint256 voterStake = token.balanceOfAt(_voter, _vote.snapshotBlock);
 
-            if (!_hasCastVote(vote_, voter)) {
-                _castVote(vote_, _voteId, _supports, voter, msg.sender);
-                emit ProxyVoteSuccess(_voteId, voter, msg.sender, _supports);
-            } else {
-                emit ProxyVoteFailure(_voteId, voter, msg.sender);
+        VoteCast storage castVote = _vote.castVotes[_voter];
+        VoterState previousVoterState = castVote.state;
+
+        // If voter had previously voted, reset their vote
+        // Note that votes can only be changed once by the principal voter overruling their representative's
+        // vote during the overrule window
+        if (previousVoterState == VoterState.Yea) {
+            yeas = yeas.sub(voterStake);
+        } else if (previousVoterState == VoterState.Nay) {
+            nays = nays.sub(voterStake);
+        }
+
+        if (_supports) {
+            yeas = yeas.add(voterStake);
+        } else {
+            nays = nays.add(voterStake);
+        }
+
+        _vote.yea = yeas;
+        _vote.nay = nays;
+        castVote.state = _voterStateFor(_supports);
+        castVote.caster = _caster;
+        emit CastVote(_voteId, _voter, _supports, _caster == address(0) ? _voter : _caster);
+
+        // Note: could we move this to be at the start of `castVote` so it acts like a modifier /
+        // pre-condition, making sure we're calculating the quiet ending period correctly?
```

Comment:
Having it here allows us to work with the previous and current support values to now if the outcome was flipped actually

Reply:
Addressed in 798589a04a663ab217c717e52a6f73c81a3dc196